src/platform/src/redox/mod.rs

```diff
@@ -135,6 +137,14 @@ pub fn mkdir(path: *const c_char, mode: mode_t) -> c_int {
     }
 }
 
+pub fn nanosleep(rqtp: *const timespec, rmtp: *mut timespec) -> c_int {
+    unsafe {
+        let rqtp = mem::transmute::<*const timespec, &redox_timespec>(rqtp);
```

Comment:
I don't recommend transmuting. Instead to what I did for pipe - create a redox_timespec and copy to and from the timespec

Reply:
It's the exact same time though, just with a different name.

Reply:
Is it repr(C) ?

Reply:
Plus that would be an allocation wouldn't it?

Reply:
If done on the stack it is ok

Reply:
So it looks like we uncovered a problem with the nanosleep syscall not handling null values properly.

Reply:
https://github.com/redox-os/kernel/blob/master/src/syscall/mod.rs#L77-L84 the syscall does handle null values correctly, the implementation in `syscall::call` is just not as usable as it could be.